### PR TITLE
Chore/#173114232 remove simplejson

### PR DESCRIPTION
**Pull request by topliceanu**
### Why?

- `bitly/simplejson` offers the same functionality as `tidwall/gjson`.
- we're using gjson in a lot of places, but simplejson only in a single file. 

### How?

- I've update `core/adaptors/json_parser.go`. The codebase is that file could be improved some more.

### CODE REVIEW

File: core/adapters/json_parse.go [hunk 1]
```diff
@@ -52,26 +52,27 @@ func (jpa *JSONParse) Perform(input models.RunInput, _ *store.Store) models.RunO
 		return models.NewRunOutputError(err)
 	}
 
-	js, err := simplejson.NewJson([]byte(val))
```
Comment (topliceanu) [inline core/adapters/json_parse.go:55]: `input.Result()` is of type gjson.Result already. I made the assumption that the input was not previously checked to be correctly encoded as JSON.  I can remove this line if that is not the case.

File: core/adapters/json_parse.go [hunk 2]
```diff
@@ -52,26 +52,27 @@ func (jpa *JSONParse) Perform(input models.RunInput, _ *store.Store) models.RunO
...
-	if err != nil {
-		return models.NewRunOutputError(err)
+	if !gjson.Valid(val) {
+		return models.NewRunOutputError(errors.New("invalid json input"))
 	}
+	js := gjson.Parse(val)
```
Comment (topliceanu) [inline core/adapters/json_parse.go:58]: Even though `input.Result()` is of type gjson.Result, the input is sometimes encoded twice - that's certainly the case in the tests. I have taken the safer route to parse the input again using gjson. It's not very nice, I susspect it is related to these lines https://github.com/smartcontractkit/chainlink/pull/3038/files#diff-4cf19e75aa397a77b9d2d738a7d184c4R44-R49

File: core/adapters/json_parse.go [hunk 3]
```diff
@@ -82,37 +83,19 @@ func dig(js *simplejson.Json, path []string) (*simplejson.Json, error) {
...
-func moldErrorOutput(js *simplejson.Json, path []string, input models.RunInput) models.RunOutput {
-	if _, err := getEarlyPath(js, path); err != nil {
+func moldErrorOutput(js gjson.Result, path []string) models.RunOutput {
+	if _, err := dig(js, path[:len(path) - 1]); err != nil {
 		return models.NewRunOutputError(err)
 	}
 	return models.NewRunOutputCompleteWithResult(nil)
 }
 
-func getEarlyPath(js *simplejson.Json, path []string) (*simplejson.Json, error) {
```
Comment (topliceanu) [inline core/adapters/json_parse.go:92]: this function is a copy of `dig()` with the different that the input path is a sub-slice. I refactored to use dig directly.

**Comment by topliceanu** [comment]
CC @samsondav

File: core/adapters/json_parse.go (hunk 2 above)
Comment (samsondav) [reply core/adapters/json_parse.go:57]: Yeah this part of the code is a bit of a nightmare.

File: core/adapters/json_parse.go (hunk 1 above)
Comment (samsondav) [reply core/adapters/json_parse.go:55]: Seems like the safe option for now.

**Pull request merged**